### PR TITLE
Text on photo

### PR DESCRIPTION
**Pull request by dyxcooh**
Add any text on photo: after tapping the desired text on top of the screen and choosing the color, click on "Add Text" button.
The text appeared on photo can be dragged by one finger, rotate or zoom using two fingers.
To edit the text: modify the content on top of the screen and pick new color, then long press on the text you want to edit.

### CODE REVIEW

**Comment by aman3014** [comment]
Thanks a lot for this work! Could you `git pull origin main` and resolve conflicts before we review please?

**Comment by dyxcooh** [comment]
> Thanks a lot for this work! Could you merge main and resolve conflicts before we review please?

Yes, it's done!

File: app/src/main/java/com/github/epfl/meili/photo/PhotoEditActivity.kt [hunk 1]
```diff
@@ -57,19 +59,24 @@ class PhotoEditActivity : AppCompatActivity() {
...
+            override fun onAddViewListener(p0: ViewType?, p1: Int) {} override fun onRemoveViewListener(p0: ViewType?, p1: Int) {}
+            override fun onStartViewChangeListener(p0: ViewType?) {} override fun onStopViewChangeListener(p0: ViewType?) {} })
+
+        // photo will get stored on phone once it is done being edited
+        if (!isPermissionGranted()) { getStoragePermission() }
         setFabListener()
 
         // Handle emojis
-        binding.emojis.setOnClickListener { showEmojiTable() }
+        /** Show scrollable emoji table */
```
Comment (Bwananah) [inline app/src/main/java/com/github/epfl/meili/photo/PhotoEditActivity.kt:74]: Nitpick : this comment is a bit weird now that it's not above a function anymore. Maybe you could delete it or make it // ?

File: app/src/main/java/com/github/epfl/meili/photo/PhotoEditActivity.kt [hunk 2]
```diff
@@ -137,39 +162,39 @@ class PhotoEditActivity : AppCompatActivity() {
...
-        binding.emojiContainer.visibility = View.VISIBLE
-    }
-
-
-    private fun changeDrawingColor() {
-        photoEditor.brushColor = binding.colorSlider.color
-    }
+    //private fun changeDrawingColor() {
+    //    photoEditor.brushColor = binding.colorSlider.color
+    //}
```
Comment (Bwananah) [inline app/src/main/java/com/github/epfl/meili/photo/PhotoEditActivity.kt:197]: This is commented code, do we have to keep it here ?

**Review by Bwananah — APPROVED** [review]
Good job ! I have a few nitpicks, but they only concern comments, so its not a big deal.
Very well made, thanks for implementing it !

File: app/src/main/java/com/github/epfl/meili/photo/PhotoEditActivity.kt (hunk 2 above)
Comment (dyxcooh) [reply app/src/main/java/com/github/epfl/meili/photo/PhotoEditActivity.kt]: I thought maybe someone will want to reuse it, but anyway it's in git's history so we can remove it.

**Review by aman3014 — APPROVED** [review]
Thanks for this feature!

**Pull request merged**